### PR TITLE
Improve Incremental Builds

### PR DESCRIPTION
**Pull request by mpm-os**
View Core Generation always runs if we change a `.tsx `or a `.view.tsx`.

With this fix it will only run if we change a `view.tsx`.

### CODE REVIEW

File: ViewGeneratorCore/build/ViewGeneratorCore.targets [hunk 1]
```diff
@@ -39,7 +39,7 @@
   </Target>
 
   <ItemGroup>
-    <VGC_Inputs Include="$(OverrideProjectDir)**\*.tsx" Exclude="$(OverrideProjectDir)node_modules\**\*.tsx" />
+    <VGC_Inputs Include="$(OverrideProjectDir)**\*.view.tsx" Exclude="$(OverrideProjectDir)node_modules\**\*.tsx" />
```
Comment (joaompneves) [inline ViewGeneratorCore/build/ViewGeneratorCore.targets:42]: view.tsx... is just a convention... cant be used here

**Review by joaompneves — CHANGES_REQUESTED** [review]

File: ViewGeneratorCore/build/ViewGeneratorCore.targets (hunk 1 above)
Comment (mpm-os) [reply ViewGeneratorCore/build/ViewGeneratorCore.targets]: fixed, this pattern is now parsed from ts2lang.json config file

File: ViewGeneratorCore/build/ViewGeneratorCore.targets [hunk 2]
```diff
@@ -36,13 +36,33 @@
       <__ViewGeneratorCoreNodeJsExe>@(_ViewGeneratorCoreNodeJsExeAfterCheckedPaths);</__ViewGeneratorCoreNodeJsExe>
       <ViewGeneratorCoreNodeJsExe>$(__ViewGeneratorCoreNodeJsExe.Substring(0, $(__ViewGeneratorCoreNodeJsExe.IndexOf(';'))))</ViewGeneratorCoreNodeJsExe>
     </PropertyGroup>
+
+      <PropertyGroup>
```
Comment (joaompneves) [inline ViewGeneratorCore/build/ViewGeneratorCore.targets:40]: have you tested this in ss?

Comment (mpm-os) [reply ViewGeneratorCore/build/ViewGeneratorCore.targets:40]: Yes, this seems to be working fine. Do you have any concern about this?

Comment (joaompneves) [reply ViewGeneratorCore/build/ViewGeneratorCore.targets:40]: the regex scares me a bit

Comment (mpm-os) [reply ViewGeneratorCore/build/ViewGeneratorCore.targets:40]: I understand, any suggestions? new config file?

I don't know if we can parse JSON on msbuild without external packages, that way we could access the input key without regex ..

File: ViewGeneratorCore/build/ViewGeneratorCore.targets [hunk 3]
```diff
@@ -1,115 +1,99 @@
 ﻿<Project>
```
Comment (joaompneves) [inline ViewGeneratorCore/build/ViewGeneratorCore.targets:1]: wow such changes
much differences

Comment (mpm-os) [reply ViewGeneratorCore/build/ViewGeneratorCore.targets:1]: 😠

File: ViewGeneratorCore/build/ViewGeneratorCore.targets [hunk 4]
```diff
@@ -38,13 +38,16 @@
...
 
-  <ItemGroup>
-    <VGC_Inputs Include="$(OverrideProjectDir)**\*.tsx" Exclude="$(OverrideProjectDir)node_modules\**\*.tsx" />
-    <VGC_Inputs Include="$(OverrideProjectDir)**\*.ts" Exclude="$(OverrideProjectDir)node_modules\**\*.ts" />
-    <UpToDateCheckInput Include="@(VGC_Inputs)" Exclude="@(UpToDateCheckInput)" />
-  </ItemGroup>
-
-  <Target Name="VGC_GenerateView" Inputs="@(VGC_Inputs)" Outputs="VGC_Build.cache" Condition="'$(UsingMicrosoftNETSdk)' != 'true' Or '$(DesignTimeBuild)' != 'true'" BeforeTargets="CustomTransformDuringBuild;TransformDuringBuild;AfterResolveReferences" DependsOnTargets="VGC_SetVariables;VGC_Setup">
+    <Target Name="VGC_GetInputs" DependsOnTargets="VGC_SetVariables;VGC_Setup">
+        <Exec ConsoleToMSBuild="True" Command="&quot;$(ViewGeneratorCoreNodeJsExe)&quot; &quot;$(MSBuildThisFileDirectory)..\tools\node_modules\@outsystems\ts2lang\ts2lang-main.js&quot; -l -f &quot;$(ProjectDir)ts2lang.json&quot; -t &quot;$(MSBuildThisFileDirectory)..\tools\ViewGenerator.js&quot;">
```
Comment (joaompneves) [inline ViewGeneratorCore/build/ViewGeneratorCore.targets:42]: \ not unix compliant

**Review by joaompneves — CHANGES_REQUESTED** [review]

File: ViewGeneratorCore/build/ViewGeneratorCore.targets (hunk 4 above)
Comment (mpm-os) [reply ViewGeneratorCore/build/ViewGeneratorCore.targets:42]: This is very strange. This is working on ts2lang without those slashes, msbuild seems to be converting it.

Look at this test I implemented.
image

Comment (joaompneves) [reply ViewGeneratorCore/build/ViewGeneratorCore.targets:42]: @fernandomarquesgarcia do you know which targets need the \ properly set?

Comment (mpm-os) [reply ViewGeneratorCore/build/ViewGeneratorCore.targets:42]: hi, any update on this?

Comment (fernandomarquesgarcia) [reply ViewGeneratorCore/build/ViewGeneratorCore.targets:42]: Hi,
From what I could come up with by testing with a simple console project having one target that prints a message:
-  <Message Importance="high" Text="c:\ola.txt"/> --> **Prints "c:\ola.txt"**
- <Message Importance="high" Text="$(OutDir)c:\ola.txt" /> --> **Prints bin/Debug/net5.0/c:/ola.txt**
-    <Target Name="SetMyVars">
        <PropertyGroup>
            <MyTestPath>c:\ole.txt</MyTestPath>
        </PropertyGroup>
    </Target>
    <Target Name="Test" AfterTargets="BeforeBuild" DependsOnTargets="SetMyVars">
        <Message Importance="high" Text="$(MyTestPath)c:\ola.txt" />
    </Target>
This prints **c:\ole.txtc:\ola.txt**
-    <Target Name="SetMyVars">
        <PropertyGroup>
            <MyTestPath>$(OutDir)c:\ole.txt</MyTestPath>
        </PropertyGroup>
    </Target>
    <Target Name="Test" AfterTargets="BeforeBuild" DependsOnTargets="SetMyVars">
        <Message Importance="high" Text="$(MyTestPath)c:\ola.txt" />
    </Target>
THIS Prints **bin/Debug/net5.0/c:/ole.txtc:/ola.txt**

On file handling system targets like Copy or Include, no matter what path separator we use, msbuild translates it, so this exemple works:
**<Copy SourceFiles="test//\\\Program2.cs" DestinationFiles="test\\\\///\Program5.cs"></Copy>**

**So Summing up:**
-Attibutes that are supposed to have paths (like Source Files on Copy or Include) always translate the paths according to host system.
-Other atributes (like Text on Message) will translate the path separator if used in conjunction with a MSBUild Macro that returns a path (On Mac concatenating a path with $(ProjectDir) translates the path but doing it with $(FxCopDir) doesn't work because the macro doesn't return a value )

My Advise:
Always use slash (/) because it works on both platforms although a backslash will be translated in innumerous cases.

Comment (mpm-os) [reply ViewGeneratorCore/build/ViewGeneratorCore.targets]: thanks, changed!

**Review by joaompneves — APPROVED** [review]

**Review by joaompneves — APPROVED** [review]
